### DISCUSSION TITLE
Dataflow build taking too long

### DISCUSSION THREAD

**Discussion by Adityasrinivas24** [Xilinx/finn#1057] [Q&A]
Hi, 
I customised [CNV(w8a8)](https://github.com/Xilinx/brevitas/blob/master/src/brevitas_examples/bnn_pynq/models/CNV.py) for weight and activation bit width of 8 and tried building stiched ips, bitstream and pynq driver [as shown here](https://github.com/Xilinx/finn/blob/main/notebooks/end2end_example/bnn-pynq/cnv_end2end_example.ipynb). But the `step_hw_ipgen` generation step is taking unbelievably long to finish and I'm aware that `step_set_fifo_depths` takes longer. 

What parameters i should consider tuning for a faster build? Once a benchmark has been set, I plan on fine tuning for better performance

**Comment by fpjentzsch** [comment]
Hi,
8 bit precision is a lot for the FINN back-end and might produce a very large and time consuming design. Maybe you can start from lower bit widths and only increase them in layers where it is absolutely necessary?

In general, setting the mem_mode of your layers (e.g. MVAU) to internal_decoupled should result in faster HLS than internal_embedded (const). Switching to the new RTL components introduced in the latest FINN release might also reduce build time significantly because no HLS has to take place.
